narwhals/series.py

```diff
@@ -4001,3 +4001,63 @@ def convert_time_zone(self, time_zone: str) -> Series:
         return self._narwhals_series._from_compliant_series(
             self._narwhals_series._compliant_series.dt.convert_time_zone(time_zone)
         )
+
+    def timestamp(self, time_unit: Literal["ns", "us", "ms"] = "us") -> Series:
```

Comment:
Heads up if merging with main branch:

```suggestion
    def timestamp(self: Self, time_unit: Literal["ns", "us", "ms"] = "us") -> T:
```

(and same in Expr)